src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/InternationalPostalService.java

```diff
@@ -0,0 +1,216 @@
+            }
+            case INTERVENER2 -> {
+                IntervenerWrapper intervenerWrapper = convertToIntervener2Wrapper(caseData.get(INTERVENER_TWO));
+                isInternational = isIntervenerResideOutsideOfUK(intervenerWrapper);
+            }
+            case INTERVENER3 -> {
+                IntervenerWrapper intervenerWrapper = convertToIntervener3Wrapper(caseData.get(INTERVENER_THREE));
+                isInternational = isIntervenerResideOutsideOfUK(intervenerWrapper);
+            }
+            case INTERVENER4 -> {
+                IntervenerWrapper intervenerWrapper = convertToIntervener4Wrapper(caseData.get(INTERVENER_FOUR));
+                isInternational = isIntervenerResideOutsideOfUK(intervenerWrapper);
+            }
+            default -> isInternational = false;
+        }
+        return isInternational;
+    }
+
+    public boolean isRecipientResideOutsideOfUK(FinremCaseData caseData, String recipient) {
+        String party = String.valueOf(recipient);
+        String addressee = ObjectUtils.isNotEmpty(party) ?  WordUtils.capitalizeFully(party) : party;
+        boolean isInternational;
+        switch (addressee) {
+            case APPLICANT -> isInternational = isApplicantResideOutsideOfUK(caseData);
+            case RESPONDENT -> isInternational = isRespondentResideOutsideOfUK(caseData);
+            case INTERVENER1 -> {
+                IntervenerWrapper intervenerWrapper = caseData.getIntervenerOneWrapper();
+                isInternational = isIntervenerResideOutsideOfUK(intervenerWrapper);
+            }
+            case INTERVENER2 -> {
+                IntervenerWrapper intervenerWrapper = caseData.getIntervenerTwoWrapper();
+                isInternational = isIntervenerResideOutsideOfUK(intervenerWrapper);
+            }
+            case INTERVENER3 -> {
+                IntervenerWrapper intervenerWrapper = caseData.getIntervenerThreeWrapper();
+                isInternational = isIntervenerResideOutsideOfUK(intervenerWrapper);
+            }
+            case INTERVENER4 -> {
+                IntervenerWrapper intervenerWrapper = caseData.getIntervenerFourWrapper();
+                isInternational = isIntervenerResideOutsideOfUK(intervenerWrapper);
+            }
+            default -> isInternational = false;
+        }
+        return isInternational;
+    }
+
+    public boolean isApplicantResideOutsideOfUK(FinremCaseData caseData) {
+        ContactDetailsWrapper wrapper = caseData.getContactDetailsWrapper();
+        YesOrNo applicantResideOutsideUK = wrapper.getApplicantResideOutsideUK();
+        return applicantResideOutsideUK != null && applicantResideOutsideUK.equals(YesOrNo.YES);
+    }
+
+    public boolean isApplicantResideOutsideOfUK(Map<String, Object> caseData) {
+        String applicantResideOutsideUK = (String) caseData.get(APPLICANT_RESIDE_OUTSIDE_UK);
+        return applicantResideOutsideUK != null && applicantResideOutsideUK.equalsIgnoreCase("Yes");
```

Comment:
If you do "Yes".equalsIgnoreCase(applicantResideOutsideUK) then you don't need the null check. It would be good if you could use the YesOrNo enum in all the checks for consistency

Reply:
This is legacy one, that do not use YesOrNo. It uses "Yes" or "No"